File: client/java/src/main/java/com/be/mesh/client/system/MeshCluster.java
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */
package com.be.mesh.client.system;

import com.be.mesh.client.annotate.SPI;
import com.be.mesh.client.mpc.ServiceProxy;
import com.be.mesh.client.prsim.Cluster;

/**
 * @author coyzeng@gmail.com
 */
@SPI("mesh")
public class MeshCluster implements Cluster {

    private final Cluster cluster = ServiceProxy.proxy(Cluster.class);

    @Override
    public byte[] election(byte[] buff) {
        return cluster.election(buff);
    }

    @Override
    public boolean isLeader() {
        return cluster.isLeader();
    }
}

File: client/golang/system/mesh_cluster.go
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */

package system

import (
	"context"
	"github.com/opendatav/mesh/client/golang/macro"
	"github.com/opendatav/mesh/client/golang/prsim"
)

func init() {
	var _ prsim.Cluster = new(MeshCluster)
	macro.Provide(prsim.ICluster, new(MeshCluster))
}

type MeshCluster struct {
}

func (that *MeshCluster) Att() *macro.Att {
	return &macro.Att{Name: macro.MeshSys}
}
func (that *MeshCluster) Election(ctx context.Context, buff []byte) ([]byte, error) {
	return aware.Cluster.Election(ctx, buff)
}

func (that *MeshCluster) IsLeader(ctx context.Context) (bool, error) {
	return aware.Cluster.IsLeader(ctx)
}
